eslint.config.mjs

```diff
@@ -0,0 +1,94 @@
+// @ts-check
+
+import path from 'node:path';
+import {fileURLToPath} from 'node:url';
+import react from '@eslint-react/eslint-plugin';
+import {includeIgnoreFile} from '@eslint/compat';
+import js from '@eslint/js';
+import pluginQuery from '@tanstack/eslint-plugin-query';
+// @ts-expect-error Requires updating tsconfig (see https://github.com/typescript-eslint/typescript-eslint/issues/7284)
+import * as tsParser from '@typescript-eslint/parser';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+const gitignorePath = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  '.gitignore',
+);
+
+const gitExcludePath = path.join(
+  path.dirname(fileURLToPath(import.meta.url)),
+  '.git',
+  'info',
+  'exclude',
+);
+
+const toolingConfig = tseslint.config({
+  name: 'tooling',
+  files: [
+    '*.config.{js,mjs,cjs}',
+    'scripts/*.{js,mjs,cjs}',
+    'expo-config-plugins/*.{js,mjs,cjs}',
+  ],
+  languageOptions: {
+    globals: {
+      ...globals.node,
+      ...globals.nodeBuiltin,
+      ...globals.worker,
+    },
+  },
+});
+
+const backendConfig = tseslint.config({
+  name: 'backend',
+  files: ['src/backend/**/*.{js,ts}'],
+  extends: [tseslint.configs.recommended],
+  languageOptions: {
+    globals: {
+      ...globals.node,
+      ...globals.nodeBuiltin,
+      ...globals.worker,
+    },
+  },
+});
+
+// TODO: Incorporate react-native
+const frontendConfig = tseslint.config({
+  name: 'frontend',
+  files: ['src/frontend/**/*.{js,jsx,ts,tsx}'],
+  extends: [
+    tseslint.configs.recommended,
+    pluginQuery.configs['flat/recommended'],
+    react.configs['recommended-typescript'],
+    react.configs['disable-dom'],
```

Comment:
should we also disable the `web-api`?

Reply:
I think the `web-api` has some useful rules that are still relevant to React Native code (e.g. https://eslint-react.xyz/docs/rules/web-api-no-leaked-interval), although maybe it's worth disabling a rule like https://eslint-react.xyz/docs/rules/web-api-no-leaked-resize-observer. i'm on the fence about https://eslint-react.xyz/docs/rules/web-api-no-leaked-event-listener because it's generally a helpful rule but does add some noise due to some apis using the `const subscription = foo.addEventListener()` pattern. If the noise is minimal compared to the utility, I'm currently in favor of adding ignore comments where the rule is complaining as opposed to disabling the rule altogether.